src/pycram/designators/specialized_designators/action/dual_arm_pickup_action.py

```diff
@@ -0,0 +1,88 @@
+        """
+        Specialized version of the PickUpAction designator which uses heuristics to solve for a dual pickup problem. The
+        designator will choose the arm which is closest to the object that is to be picked up.
+        :param object_designator_description: List of object designator which should be picked up
+        :param grasps: List of possible grasps which should be used for the pickup
+        :param resolver: Optional specialized_designators that returns a performable designator with elements from the
+                         lists of possible parameter
+        :param ontology_concept_holders: List of ontology concepts that the action is categorized as or associated with
+        """
+        super().__init__(object_designator_description,
+                         arms=[Arms.LEFT, Arms.RIGHT],
+                         grasps=grasps,
+                         resolver=resolver,
+                         ontology_concept_holders=ontology_concept_holders)
+
+        self.object_designator_description: Union[
+            ObjectDesignatorDescription, ObjectDesignatorDescription.Object] = object_designator_description
+
+        left_gripper_frame = World.robot.get_link_tf_frame(
+            RobotDescription.current_robot_description.kinematic_chains.get('left').get_tool_frame())
+        right_gripper_frame = World.robot.get_link_tf_frame(
+            RobotDescription.current_robot_description.kinematic_chains.get('right').get_tool_frame())
+        self.gripper_list: List[str] = [left_gripper_frame, right_gripper_frame]
+
+
+    def ground(self) -> PickUpActionPerformable:
+        if isinstance(self.object_designator_description, ObjectDesignatorDescription.Object):
+            obj_desig = self.object_designator_description
+        else:
+            obj_desig = self.object_designator_description.resolve()
+
+        rospy.loginfo("Calculating closest gripper to object {}".format(obj_desig.name))
+
+        local_transformer = LocalTransformer()
+
+        object_frame = obj_desig.world_object.tf_frame
+        distances = []
+        # Iterate over possible grippers
+        for gripper in self.gripper_list:
+            # Object pose in gripper frame
+            object_T_gripper: Transform = local_transformer.lookup_transform_from_source_to_target_frame(gripper,
+                                                                                            object_frame)
+            object_V_gripper: Vector3 = object_T_gripper.translation  # translation vector
+            distance = norm(array([object_V_gripper.x, object_V_gripper.y, object_V_gripper.z]))
+            rospy.loginfo(f"Distance between {gripper} and {obj_desig.name}: {distance}")
+            distances.append(distance)
+
+        min_index = distances.index(min(distances))
+        winner_frame = self.gripper_list[min_index]
+
+        if winner_frame == World.robot.get_link_tf_frame(
```

Comment:
The kinematic chain description contains the arm_type (i.e. Arms.LEFT or Arms.RIGHT) it would be better to use this to avoid this if-else block. You would need to save the KinemacticChainDescriptions in the gripper_list, though